### PR TITLE
Fixed a couple of issues

### PR DESCRIPTION
**Pull request by ghpotter**
Dear Olivier,

I worked with Chris Swierczewski this spring on updating some of the pychebfun code.  I finally got around to finalizing my work.  I worked on issues #2 (Chebfun from Chebyshev coefficients) and #3 (implementing the DCT).  Additionally, I fixed the differentiate function.  To test the differentiate function, I removed the expected failure of the differentiate function in the test_chebfun.py file, and there were no errors when I ran it.

Best regards,
Greg Potter

### CODE REVIEW

**Comment by olivierverdier** [comment]
Very nice!!

I believe it indeed fixes #2 and #3.

File: pychebfun/chebfun.py [hunk 1]
```diff
@@ -47,14 +48,17 @@ class NoConvergence(Exception):
         Raised when dichotomy does not converge.
         """
 
-    def __init__(self, f=None, N=0,):
+    def __init__(self, f=None, N=0, chebcoeff=np.array(0),):
```
Comment (olivierverdier) [inline pychebfun/chebfun.py]: I think a better default value would be None (mutable default arguments are not really safe in Python)

File: pychebfun/chebfun.py [hunk 2]
```diff
@@ -84,42 +88,62 @@ def __init__(self, f=None, N=0,):
             self.x = f.x
             self.f = f.f
 
-        if not N: # N is not provided
-            # Find out the right number of coefficients to keep
-            for k in xrange(2,self.max_nb_dichotomy):
-                N = pow(2,k)
+        Chebflag = False
+        try:
+            if chebcoeff.any():
```
Comment (olivierverdier) [inline pychebfun/chebfun.py]: Would then be replaced by `if chebcoeff is not None:`

File: pychebfun/chebfun.py [hunk 3]
```diff
@@ -84,42 +88,62 @@ def __init__(self, f=None, N=0,):
...
+                Chebflag = True
+        except:
+            print "Chebyshev coefficients should be in an array.\n"
```
Comment (olivierverdier) [inline pychebfun/chebfun.py]: I think it should raise an exception instead of printing something; `raise TypeError(...)` (or `ValueError`, I don't know)

**Comment by olivierverdier** [comment]
There are really two different things in that one commit, right?

On the one hand, the fft->dct, and that seems fine. Also, all the current test already implicitly test that.

On the other hand, the possibility to initialise the chebfun with coefficients. It looks good with the `test_differentiate` so far, but I think it would not hurt to have more tests for that.

File: pychebfun/chebfun.py (hunk 3 above)
Comment (olivierverdier) [inline pychebfun/chebfun.py]: In fact, I suppose a better solution would be to remove Chebflag alltogether, and to replace `if Chebflag` by `if chebcoeff is not None`.

Later we should also move some of those initialisation bits in separate function, both for better testing, and to make that long function `__init__` easier to read.

**Comment by ghpotter** [comment]
Sorry, I should have done different pull requests for the different issues.

I implemented some of the changes.  Namely, I removed the Chebflag marker and just did the "if chebcoeff is not None" and removed the error message.  Also, I wrote a separate function for inverse DCT.

I'm not sure, exactly how updating my pull request works, but I pushed my current code with this fixes.

**Comment by olivierverdier** [comment]
Brilliant! It is better this way. However, I see that the `test_differentiate` test does not pass anymore...

Also, I think that some new tests to check the chebcoeff argument, are in order. I'll try and think of something.

**Comment by olivierverdier** [comment]
Ok, I have now reorder your commits and added tests for each commit.

I improved the differentiating test (there was hardly any before), and it doesn't pass. I'm not sure if it is an implementation error, or if it is just supposed to be like that... See Issue #9.

Thanks a lot! Feel free to tackle some other issues! :-)

(Note: you should now checkout my branch and `git reset --hard master`, because I cherry-picked parts of your commits)

**Pull request closed without merging**